src/Client/Stripe.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BigGive\Identity\Client;
+
+use BigGive\Identity\Tests\Client\Stripe\StubCustomerService;
+use Stripe\StripeClient;
+
+class Stripe
+{
+    private ?StripeClient $stripeNativeClient = null;
+
+    public function __construct(bool $stubbed, array $stripeOptions)
+    {
+        if ($stubbed && getenv('APP_ENV') === 'production') {
+            throw new \LogicException('Cannot stub out Stripe in production');
+        }
+
+        if (!$stubbed) {
+            $this->stripeNativeClient = new StripeClient($stripeOptions);
+            return;
+        }
+
+        // Else, fake everything we must for load tests. Anything not stubbed explicitly will crash in
+        // stub mode.
+        $this->customers = new StubCustomerService();
+    }
+
+    public function __get($name)
```

Comment:
```suggestion
    public function __get(string $name)
```

Reply:
And of course let's not keep using `__get` long term here - makes static analysis pretty much impossible, and the fact that its there means Psalm won't alert us if we have a typo like `$this->stripeClient->ccustomers`